### ISSUE TITLE
more than one entry needed for surface forcing

### ISSUE THREAD

**Issue by dafyddstephenson** [CWorthy-ocean/roms-tools#89]
```
roms_frc_bgc= rt.SurfaceForcing(
    grid=roms_grd,
    start_time=dt.datetime(2012, 1, 1, 0, 0, 0),
    end_time=dt.datetime(2012, 2, 1, 0, 0, 0),
    physics_source={"name": "ERA5", "path": datadir+'ERA5*.nc'},
    bgc_source={"name": "CESM_REGRIDDED", "path": datadir+"CESM-surface_lowres_regridded.nc"},
    correct_radiation=True,
    use_coarse_grid=True,
)
roms_frc.save(filepath='roms_frc')
```
This generates two forcing files, as expected. The issue I experience is that the physics file has 744 entries under time, while the BGC file has just 1. While this is consistent with the data that went in to generating the forcing, I believe ROMS needs at least two entries to interpolate between, and so aborts on initialisation:
` ERROR: find_new_record: Ran out of time records for dust `

Summary:
Forcing objects should always have at least two time entries.

**Comment by dafyddstephenson** [comment]
Related: #82

**Comment by NoraLoose** [comment]
Thanks for raising this @dafyddstephenson! Do the two time entries have to fully enclose the ROMS simulation? Or will ROMS extrapolate?

**Comment by dafyddstephenson** [comment]
I'm quite certain ROMS will interpolate between two values 1000+ years apart but will not extrapolate...

**Comment by NoraLoose** [comment]
So if you want to run your simulation from January 1 to February 1, we can't just give it two values on January 15 and February 15. Instead we need three values: on December 15, January 15, February 15?

**Comment by dafyddstephenson** [comment]
I think we can give it Jan 1st and Feb 1st. If we only have 15ths available then yes it would be Dec 15th of the previous year to Feb 15th of the current year.

However the one value I am currently getting in my forcing file corresponds to January 1st 2012, not January 15th

**Comment by NoraLoose** [comment]
Solved via #100

**Comment by NoraLoose** [comment]
I'm opening this issue again because there is some unexpected / unwanted behaviour with the time selection:
```
start_time = datetime(2012, 8, 1)
end_time = datetime(2012, 8, 7)
surface_forcing = SurfaceForcing(
    grid=grid,
    start_time=start_time,
    end_time=end_time,
    source={"name": "ERA5", "path": era5_path},
    correct_radiation=True, 
    type="physics",
)
```
leads to `surface_forcing.ds` having the time entries
```
array(['2012-08-01T00:00:00.000000000', '2012-08-01T01:00:00.000000000',
       '2012-08-01T02:00:00.000000000', '2012-08-01T03:00:00.000000000',
        ...
       '2012-08-06T22:00:00.000000000', '2012-08-06T23:00:00.000000000',
       '2012-08-07T00:00:00.000000000'], dtype='datetime64[ns]')
```

It should return time entries all the way to
```
       '2012-08-08T00:00:00.000000000'], dtype='datetime64[ns]')
```
